src/button/types.js

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import {KIND, SIZE, SHAPE} from './constants';
+import type {OverrideT} from '../helpers/overrides';
+
+export type OverridesT = {
+  BaseButton?: OverrideT<*>,
+  StartEnhancer?: OverrideT<*>,
+  EndEnhancer?: OverrideT<*>,
+};
+
+export type ButtonPropsT = {
+  startEnhancer?: () => React$Node | React$Node,
+  endEnhancer?: () => React$Node | React$Node,
```

Comment:
Why the double `React$Node`?

Also, can you educate me on what the difference between React$Node and React.Node from `import * as React`? Seems like we use both in various places.

Reply:
afaik , React.Node comes from React files itself, whereas React$Node may come from flow-typed folder ( I don't see it, but maybe somewhere in deeper imports)
To use React.Node we need to import like this only:
`import React from 'react';`
https://flow.org/en/docs/react/types/#toc-react-node

Reply:
Makes sense. Curious if there is a "best practice" for which one to use.

Reply:
So it's either a function that returns a `React$Node` here or a `React$Node` itself. I was trying to follow the convention from `Input` here. Though maybe we should also talk about what we really expect for these `enhancers`?

Reply:
this one is more common, it includes them both, function also
https://github.com/facebook/flow/blob/v0.79.1/lib/react.js#L152

Reply:
You mean we should switch to accepting components for `enhancers`?

Reply:
no, just use flow type, which is more common